### PR TITLE
statistics: refactor write merge-global-stats

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### What changed and how does it work?

In the future, we will add concurrently global stats writing here.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/statistics/handle/globalstats/write_result.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package globalstats
+
+import (
+	statslogutil "github.com/pingcap/tidb/pkg/statistics/handle/logutil"
```
Comment (AilinKid) [inline pkg/statistics/handle/globalstats/write_result.go:18]: seem no alias is ok as well

**Review by AilinKid — APPROVED** [review]

File: pkg/statistics/handle/types/interfaces.go [hunk 2]
```diff
@@ -418,15 +418,23 @@ type StatsSyncLoad interface {
 	HandleOneTask(sctx sessionctx.Context, lastTask *NeededItemTask, exit chan struct{}) (task *NeededItemTask, err error)
 }
 
+// GlobalStatsInfo represents the contextual information pertaining to global statistics.
+type GlobalStatsInfo struct {
+	// When the `isIndex == 0`, HistIDs will be the column IDs.
```
Comment (hi-rustin) [inline pkg/statistics/handle/types/interfaces.go:423]: Could you please move the comment to the corresponding field?

File: pkg/statistics/handle/globalstats/write_result.go
Comment (hi-rustin) [inline pkg/statistics/handle/globalstats/write_result.go:1]: The file name is quite confusing.

File: pkg/statistics/handle/types/interfaces.go (hunk 2 above)
Comment (hawkingrei) [reply pkg/statistics/handle/types/interfaces.go:424]: make sense

File: pkg/statistics/handle/globalstats/write_result.go (hunk 1 above)
Comment (hawkingrei) [reply pkg/statistics/handle/globalstats/write_result.go]: Done

File: pkg/statistics/handle/globalstats/write_result.go
Comment (hawkingrei) [reply pkg/statistics/handle/globalstats/write_result.go:1]: ```write_merge_global.go```?

Comment (hi-rustin) [reply pkg/statistics/handle/globalstats/write_result.go:1]: How about we keep it in the global_stats.go?

Comment (hawkingrei) [reply pkg/statistics/handle/globalstats/write_result.go:1]: global_stats is to handle the merge_global_stats. It is best to separate the writing logic.

Comment (hawkingrei) [reply pkg/statistics/handle/globalstats/write_result.go:1]: Done, it is in the ```global_stats.go```.

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**